sinon/lib/spy.py

```diff
@@ -187,36 +179,43 @@ def lastCall(self): #pylint: disable=invalid-name
         lastIndex = len(self.__get_wrapper().call_list) - 1
         return self.getCall(lastIndex)
 
-    def calledBefore(self, obj): #pylint: disable=invalid-name
-        """
-        Comparing two spys' functions order in CALLQUEUE
-        Eg.
-            a()
-            b()
-            spy_a.calledBefore(spy_b) will return True, because a is called before b
-            a()
-            spy_b.calledBefore(spy_a) will return True, because a is called after b
-        Return: Boolean
+    def calledBefore(self, spy): #pylint: disable=invalid-name
         """
-        idx = self.get_callqueue_idx()
-        idx2 = obj.get_callqueue_idx()
+        Compares the order in which two spies were called
 
-        if len(idx) == 0:
-            ErrorHandler.callqueue_is_empty_error()
-        if Wrapper.CALLQUEUE:
-            return True if min(idx) < max(idx2) else False
+        E.g.
+            spy_a()
+            spy_b()
+            spy_a.calledBefore(spy_b) # True
+            spy_b.calledBefore(spy_a) # False
+            spy_a()
+            spy_b.calledBefore(spy_a) # True
 
-    def calledAfter(self, obj): #pylint: disable=invalid-name
+        Args: a Spy to compare with
+        Return: Boolean True if this spy's first call was called before the given spy's last call
         """
-        Same as calledBefore
+        thisCall = self.firstCall or False
+        givenCall = spy.lastCall or False
+        return (thisCall and not givenCall) or (thisCall and givenCall and thisCall.callId < givenCall.callId)
+
+    def calledAfter(self, spy): #pylint: disable=invalid-name
         """
-        idx = self.get_callqueue_idx()
-        idx2 = obj.get_callqueue_idx()
+        Compares the order in which two spies were called
+
+        E.g.
+            spy_a()
+            spy_b()
+            spy_a.calledAfter(spy_b) # False
+            spy_b.calledAfter(spy_a) # True
+            spy_a()
+            spy_a.calledAfter(spy_b) # True
 
-        if len(idx) == 0:
-            ErrorHandler.callqueue_is_empty_error()
-        if Wrapper.CALLQUEUE:
-            return True if max(idx) > min(idx2) else False
+        Args: a Spy to compare with
+        Return: Boolean True if this spy's last call was called after the given spy's first call
+        """
+        thisCall = self.lastCall or False
+        givenCall = spy.firstCall or False
+        return thisCall and givenCall and thisCall.callId > givenCall.callId
```

Comment:
Same as [above](https://github.com/note35/sinon/pull/12/files#r144797762), I suggest to put some comment about the logic.